packages/vm/src/runTx.ts

```diff
@@ -506,6 +506,11 @@ async function _runTx(this: VM, opts: RunTxOpts): Promise<RunTxResult> {
     const keys = Object.keys(results.execResult.selfdestruct)
     for (const k of keys) {
       const address = new Address(hexToBytes(k))
+      if (this._common.isActivatedEIP(6780)) {
+        if (!results.execResult.createdAddresses![address.toString()]) {
+          continue
+        }
+      }
```

Comment:
Oh, this is interesting, that this is the "only" change on the procedural code (apart from the address tracking) to get this working/implemented. 🤓 🙂